src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameOfT.cs

```diff
@@ -37,8 +37,14 @@ public class Frame<TContext> : Frame
                     {
                         if (SocketInput.RemoteIntakeFin)
                         {
+                            if (TakeStartLine(SocketInput))
```

Comment:
Please add comments describing the changes like @cesarbs suggests above.  In the future it will be hard to understand the code otherwise.